hmcts/pre-api
# S28 2881: Start Live Event

### JIRA ticket(s)
- S28-2881


### Change description
- Added endpoint `PUT /media-services/live-event/start/{captureSessionId}`
- Add start live event for AMS
- Add start live event for MK which temporarily sets ingest address to RTMP
- Start live event for both AMS and MK follow the same structure:
  - Check capture session exists/not finished/not already started
  - Create live event 
  - Create asset
  - Create live output
  - Start live event
  - Wait for live event to start
  - Return updated capture session




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Call `PUT /media-service/live-event/start/{captureSessionId}` endpoint with capture session id that does not exist- see 404
- Call endpoint with capture session id where started at is not null- see capture session is returned
- Call endpoint with capture session id where finished at is not null- see 409 
- (NORMAL CASE) Call endpoint with capture session id where only values of id, booking id and status=INIT are set- see 200 response with additional values of status= STANDBY, started_at=NOT NULL and started_by is the user who made the request + see asset and live event are created (look at AMS)
- (ATTEMPT TO BREAK AMS REQUEST)- Attempt to get a response where AMS request fails causing an error message- using `GET /capture-sessions/{id}` see values of status=FAILURE, started_at=NOT NULL, started_by=user who made request.
  - To break AMS look at the code for try catch- breaking test: manually create live event/asset in AMS before running endpoint, see conflict error response. 
- NOTES: 
  - I would recommend deleting the live events + assets created from testing directly in AMS
  - This test should be using STAGING AMS
  - In order to use the endpoint, you must have a capture session created first (use `PUT /capture-sessions/{id}`). It should have an id an attached booking, and the status should in INITIALISING.
